extensions/ql-vscode/src/query-history.ts

```diff
@@ -611,7 +613,26 @@ export class QueryHistoryManager extends DisposableObject {
       await this.refreshTreeView();
     });
 
+    const variantAnalysisStatusUpdateSubscription = this.variantAnalysisManager.onVariantAnalysisStatusUpdated(async (variantAnalysis) => {
+      const item = this.treeDataProvider.allHistory.find(i => i.t === 'variant-analysis' && i.historyItemId === variantAnalysis.id.toString());
+      const status = variantAnalysisStatusToQueryStatus(variantAnalysis.status);
+
+      if (item) {
+        const variantAnalysisHistoryItem = item as VariantAnalysisHistoryItem;
```

Comment:
Do we also need to update the `variantAnalysis` property of the `item`?

Reply:
Right you are @koesie10. I find it confusing that we're storing both the `variantAnalysis` and properties from it (like `failureReason`).

I think the lack of a test here really shows how easy it is to miss things when the code isn't being exercised anywhere. 

It's unfortunate these methods are private so we can't add unit tests.

Reply:
If there's duplicate data between the variant analysis and the other fields, I think we have the option to remove the extra fields. When we need those values we can calculate them from the variant anlaysis.

Would be best done as a separate PR though. Could you could open an issue to look into this?

Reply:
Have added an issue 👍